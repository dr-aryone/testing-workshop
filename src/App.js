import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import ChangeName from "./pages/ChangeName/ChangeName";

import Toggle from "./pages/Toggle/Toggle";

import Counter from "./pages/Counter/Counter";

import Menu from "./components/Menu/Menu";

const Router = () => {
  return (
    <BrowserRouter>
      <Menu />
      <Switch>
        <Route path="/toggle" component={Toggle} />
        <Route path="/counter" component={Counter} />
        <Route path="/change" component={ChangeName} />
      </Switch>
    </BrowserRouter>
  );
};

export default Router;
