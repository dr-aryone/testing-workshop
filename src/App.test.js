import React from "react";
import { render, fireEvent } from "@testing-library/react";

import App from "./App";

describe("App Component", () => {
  describe("Router", () => {
    it("Should go to the Toggle Component", () => {
      const { getByText, getByTestId } = render(<App />);

      const toggleLink = getByText("Toggle");

      fireEvent.click(toggleLink);

      const toggleContainer = getByTestId("toggleContainer");

      expect(toggleContainer).toBeInTheDocument();
    });

    it("Should go to the Counter Component", () => {
      const { getByText, getByTestId } = render(<App />);

      const counterLink = getByText("Counter");

      fireEvent.click(counterLink);

      const counterContainer = getByTestId("counterContainer");

      expect(counterContainer).toBeInTheDocument();
    });

    it("Should go to the ChangeName Component", () => {
      const { getByText, getByTestId } = render(<App />);

      const changeNameLink = getByText("Change Name");

      fireEvent.click(changeNameLink);

      const changeNameContainer = getByTestId("changeNameContainer");

      expect(changeNameContainer).toBeInTheDocument();
    });
  });
});
