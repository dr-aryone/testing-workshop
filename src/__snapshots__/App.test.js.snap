// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Match Snapshot 1`] = `undefined`;
